Assets/PatchKit Patcher/Scripts/Debug/DebugMenu.cs

```diff
@@ -0,0 +1,216 @@
+using UnityEngine;
+using UnityEngine.UI;
+
+namespace PatchKit_Patcher.Scripts.Debug
+{
+    public class DebugMenu : MonoBehaviour
+    {
+        private static readonly DebugLogger DebugLogger = new DebugLogger(typeof(DebugMenu));
+
+        private Rect _popUp, _debugMenu;
+        private bool _show;
+        private bool _showPopUp;
+        private string _message;
+        private GraphicRaycaster _graphicRaycaster;
+
+        void Start()
+        {
+            int windowWidth = 250;
+            int windowHeight = 180;
+            int x = (Screen.width - windowWidth) / 2;
+            int y = (Screen.height - windowHeight) / 2;
+            int yPopUp = (Screen.height - 120) / 2;
+            _debugMenu = new Rect(x, y, windowWidth, windowHeight);
+            _popUp = new Rect(x, yPopUp, windowWidth, 120);
+            _graphicRaycaster = FindObjectOfType<GraphicRaycaster>();
+        }
+
+        void OnGUI()
+        {
+            if (Event.current.Equals(Event.KeyboardEvent(KeyCode.D.ToString())))
+            {
+                _show = !_show;
+                _graphicRaycaster.enabled = !_graphicRaycaster.enabled;
+            }
+
+            if (_show)
+            {
+                GUI.DrawTexture(_debugMenu, Texture2D.whiteTexture);
+                GUI.Window(0, _debugMenu, Menu, "Debug Menu");
+            }
+            else if (_showPopUp)
+            {
+                GUI.DrawTexture(_popUp, Texture2D.whiteTexture);
+                GUI.Window(1, _popUp, PopUp, "Information");
+            }
+        }
+
+        void Menu(int i)
+        {
+            if (GUILayout.Button("Open Patcher log file"))
+            {
+                OpenPatcherLogFile();
+            }
+
+            if (GUILayout.Button("Open Patcher log file location"))
+            {
+                OpenPatcherLogFileLocation();
+            }
+
+            if (GUILayout.Button("Open Launcher log file"))
+            {
+                OpenLauncherLogFile();
+            }
+
+            if (GUILayout.Button("Open Launcher log file location"))
+            {
+                OpenLauncherLogFileLocation();
+            }
+
+            if (Patcher.Instance.IsAppInstalled.Value)
+            {
+                if (GUILayout.Button("Verify all app files"))
+                {
+                    VerifyAllAppFiles();
+                }
+
+                if (GUILayout.Button("Remove all app files"))
+                {
+                    RemoveAllAppFiles();
+                }
+            }
+        }
+
+        private void VerifyAllAppFiles()
+        {
+            SetUserDecision(Patcher.UserDecision.VerifyFiles);
+        }
+
+        private void RemoveAllAppFiles()
+        {
+            SetUserDecision(Patcher.UserDecision.UninstallApp);
+        }
+
+        private void SetUserDecision(Patcher.UserDecision userDecision)
+        {
+            while (Patcher.Instance.State.Value != PatcherState.WaitingForUserDecision)
```

Comment:
Please convert it to coroutine as starting it in the main thread might freeze Unity due to this `while` loop.